src/NuGet.Clients/NuGet.Console/Xamls/ConsoleContainer.xaml.cs

```diff
@@ -1,21 +1,29 @@
 // Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
+using System;
+using System.Threading;
 using System.Windows;
 using System.Windows.Controls;
+using Microsoft;
+using Microsoft.ServiceHub.Framework;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
 using NuGet.PackageManagement;
 using NuGet.PackageManagement.UI;
+using NuGet.PackageManagement.VisualStudio;
 using NuGet.VisualStudio;
+using NuGet.VisualStudio.Internal.Contracts;
 
 namespace NuGetConsole
 {
     /// <summary>
     /// Interaction logic for ConsoleContainer.xaml
     /// </summary>
-    public partial class ConsoleContainer : UserControl
+    public sealed partial class ConsoleContainer : UserControl, IDisposable
     {
+        private INuGetSolutionManagerService _solutionManager;
```

Comment:
>_solutionManager [](start = 45, length = 16)

Why are we storing the proxy and passing it everywhere? We should be getting the proxy as needed and getting rid of it after its used.  If we insist on storing the proxy we need to hook the AvailabilityChanged event and refresh it in the event that the current status changes, e.g. If the user switches from a local folder to joining a Live Share session, the proxy you got before may no longer apply and you'd need to refresh it.

Reply:
`INuGetSolutionManagerService` offers remoteable equivalents of `ISolutionManager` events.  The proxy needs to live at least as long as events are subscribed to or as long as availability hasn't changed --- whichever comes first.  This is why one proxy instance is alive longer than the typical method scope.

In the latest iteration, I've added a listener for the `AvailabilityChanged` event to requery for the service.